front_end/src/components/post_card/basic_post_card/index.tsx

```diff
@@ -71,6 +78,30 @@ const BasicPostCard: FC<PropsWithChildren<Props>> = ({
         <div className="hidden lg:inline-flex">
           <PostDefaultProject defaultProject={defaultProject} />
         </div>
+        {/* This is awkward, takes logic internal to PostDefaultProject and
+        button from sidebar_question_tags.tsx. Should rework. */}
+        {globalLeaderboard &&
+          (![
+            TournamentType.Tournament,
+            TournamentType.GlobalLeaderboard,
+            TournamentType.QuestionSeries,
+          ].includes(defaultProject.type) ||
+            !defaultProject.default_permission) && (
```

Comment:
This bit here is just a check to see if PostDefaultProject returns null
Is there a cleaner way to do that?

Reply:
I'd suggest to make `PostDefaultProject` to be responsible for handling both these cases. We would probably also need to rename it to something like `ProjectBadge`. 

Here is a rough idea how that component would look like:

```
  const withDefaultProjectBadge =
    [
      TournamentType.Tournament,
      TournamentType.GlobalLeaderboard,
      TournamentType.QuestionSeries,
    ].includes(defaultProject.type) && !!defaultProject.default_permission;

  if (!withDefaultProjectBadge) {
    if (globalLeaderboard) {
      return (
        <Chip
          key={globalLeaderboard.id}
          href={`/questions/?${POST_TAGS_FILTER}=${globalLeaderboard.slug}&for_main_feed=false`}
          color="gray"
          onClick={() =>
            sendGAEvent("event", "questionTagClicked", {
              event_category: globalLeaderboard.name,
            })
          }
        >
          {globalLeaderboard.name}
        </Chip>
      );
    }

    return null;
  }

  return (
    <Link
      className="inline-flex items-center justify-center gap-1 rounded-l rounded-r border-inherit bg-orange-100 p-1.5 text-sm font-medium leading-4 text-orange-900 no-underline hover:bg-orange-200 dark:bg-orange-100-dark dark:text-orange-900-dark hover:dark:bg-orange-200-dark"
      href={`/tournament/${defaultProject.slug ?? defaultProject.id}`}
    >
      {defaultProject.name}
    </Link>
  );
```